### PR TITLE
Compile our binaries statically

### PR DESCRIPTION
**Pull request by metalmatze**
I also removed some variables from the top that were only used in one place or not used at all.

### CODE REVIEW

**Review by mrIncompetent — DISMISSED** [review]

File: api/Makefile [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-SHELL:=/bin/bash
-GO?=go
-GOBUILD=$(GO) build
-GOTEST=$(GO) test
-GOINSTALL=$(GO) install
+GO?=CGO_ENABLED=0 go
 CMD=kubermatic-api kubermatic-cluster-controller client
-GOBUILDFLAGS= -i
+GOBUILDFLAGS += -i -v
+LDFLAGS += -X main.Version=$(DRONE_TAG)
```
Comment (lbb) [inline api/Makefile]: What if this wasn't run in a Drone evironment?

**Review by lbb — CHANGES_REQUESTED** [review]

**Review by mrIncompetent — CHANGES_REQUESTED** [review]
Output from drone:
```
CGO_ENABLED=0 go build -i -v -ldflags '-w -X main.Version= -X main.Revision=8303e6a086a19807d30ab0198829029e682b9a96 -X "main.BuildDate=" -extldflags '-static'' -o _build/kubermatic-api github.com/kubermatic/kubermatic/api/cmd/kubermatic-api
```

**Comment by mrIncompetent** [comment]
Yeah seems `main.Version=` & `main.BuildDate=` are empty.
@realfake I assume thats also what will happen outside of drone. They will be simply empty & it'll compile anyway

File: api/Makefile (hunk 1 above)
Comment (metalmatze) [reply api/Makefile]: What' I actually didn't want to commit them, because we don't use them anyway. Or should we start?

**Review by lbb — APPROVED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**